demos/3d_spherical/3d_spherical.py

```diff
@@ -125,10 +115,10 @@ def layer_average(T):
 
     # Write output:
     if timestep % dump_period == 0:
-        # compute radial temperature
-        Tavg = layer_average(T)
+        # compute radially averaged temperature profile
+        averager.extrapolate_layer_average(T_avg, averager.get_layer_average(T))
         # compute deviation from layer average
-        T_dev.assign(T-Tavg)
+        T_dev.project(T-T_avg)
```

Comment:
This is probably due to the issue we discussed, but this can now be an assign again, right?